server/pulp/server/agent/direct/pulpagent.py

```diff
@@ -76,6 +77,23 @@ def cancel(context, task_id):
         admin = agent.Admin()
         admin.cancel(criteria=criteria)
 
+    @staticmethod
+    def delete_queue(url, name):
+        """
+        Purge and delete the agent queue.
+        :param url: The broker URL.
+        :type url: str
+        :param name: The queue name.
+        :type name: str
+        """
+        try:
+            queue = Queue(name)
```

Comment:
Which of these statements can raise NotFound? The others should be moved outside of the try/except block.